src/Grpc.Net.Client/Internal/StreamExtensions.cs

```diff
@@ -62,6 +62,13 @@ private static Status CreateUnknownMessageEncodingMessageStatus(string unsupport
             GrpcCallLog.ReadingMessage(call.Logger);
             cancellationToken.ThrowIfCancellationRequested();
 
+#if NET7_0_OR_GREATER
+            // Start with zero-byte read.
+            // A zero-byte read avoids renting buffer until the response is ready. Especially useful for long running streaming calls.
+            var readCount = await responseStream.ReadAsync(Memory<byte>.Empty, cancellationToken).ConfigureAwait(false);
```

Comment:
Is the common case with gRPC that the whole message will be received quickly once you start receiving data?
That is, the server won't send you 10 bytes, wait for a second, then send you the rest?

Reply:
There can be delays between messages when streaming, but a message is flushed by the server to the client all at once.